tree/arbo/tree.go

```diff
@@ -84,7 +84,7 @@ var (
 
 // Tree defines the struct that implements the MerkleTree functionalities
 type Tree struct {
-	sync.Mutex
+	sync.RWMutex
```

Comment:
remember that RWMutex is more complex and expensive than Mutex, it is only worth it if we're going to be doing mostly reads. if there is a roughly equal mix of reads and writes, a regular Mutex will perform much better.

Reply:
Right, going back to Mutex. It was part of my previous testing.